api/api.go

```diff
@@ -429,12 +441,30 @@ func (api *API) listSilences(w http.ResponseWriter, r *http.Request) {
 			}, nil)
 			return
 		}
+
+		if !matchesFilterLabels(s, matchers) {
+			continue
+		}
 		sils = append(sils, s)
 	}
 
 	respond(w, sils)
 }
 
+func matchesFilterLabels(s *types.Silence, matchers []*labels.Matcher) bool {
+	sms := map[string]string{}
+	for _, m := range s.Matchers {
```

Comment:
We probably don't want to couple our protobuf representation, which is also our internal model to that type.
I'd have to give our whole architecture a proper look and see how it all fits with the modeling changes we discussed as part of the new UI.